Ednpoints.txt

```diff
@@ -0,0 +1,147 @@
+POST: /auth/register -  registration
+
+The endpoints below will have the same logic for all roles:
+POST: /auth/login - enter to account
+POST: /auth/logout - exit from account
+POST: /auth/forgot-password - request to change password
+POST: /auth/reset-password - password reset
+
+/************/
+
+USER ENDPOINTS:
+
+GET: /users - get all users' info
+GET: /users/:userId - get user's info
+PATCH: /users/:userId - update user's info
+DELETE: /users/:userId - delete user's account
+
+GET: /dishes - get a list of dishes
+GET: /dishes/:dishId - get dish information
+GET: /dishes/popular - get popular dishes
+GET: /dishes/random - get some random dish
+
+GET: /chefs - get all chefs
+
+GET: /users/:userId/favourites/:type - get user's favourite dishes or chefs
+POST, DELETE:
+/users/:userId/favourites/:type - add/delete the dish to favourites (passing dishId)
+/users/:userId/favourites/:type - add/delete the chef to favourites (passing chefId)
+
+**team decided to store the cart data on the server so that it can be accessed when logging in from different devices, and update it using debouncing:
+GET: /users/:userId/cart - get the cart info
+POST: /users/:userId/cart - add the dish to the cart (passing dishId)
+PATCH: /users/:userId/cart - update information about the dish in the cart (passing dishId)
+DELETE: /users/:userId/cart  - delete the dish from the cart (passing dishId)
+
+**notifications will not be saved on the server
+
+GET: /orders/own - get user's orders list
+GET: /orders/:orderId - get a particular order
+POST: /orders - add a new order to user's orders list
+PATCH: /orders/:orderId - update the particular order UNTIL the chef approves it
+
+/************/
+
+CHEF ENDPOINTS:
+
+GET: /chefs/:chefId - get chef's info
+PATCH: /chefs/:chefId - update chef's info
+DELETE: /chefs/:chefId - delete chef's account
+
+GET: /dishes/own - get all dishes of this chef
+GET: /dishes/:dishId - get a particular dish
+POST: /dishes - create a new dish of this chef
+PATCH: /dishes/:dishId - update a particular dish of this chef
+DELETE: /dishes/:dishId - delete a particular dish of this chef
+
+**notifications will not be saved on the server
+
+GET: /orders/own?status=(new, completed, canceled) - get chef's all orders
+GET: /orders/:orderId - get a particular order
+PATCH: /orders/:orderId - update status of this order
+
+
+/************/
+
+COURIER ENDPOINTS:
+
+GET: /couriers?status=(active) - get the list of all delivery persons, sorted by distance and availability based on delivery person's current location and other factors
+
+GET: /couriers/:courierId - get delivery person information
+DELETE: /couriers/:courierId - delete delivery person account
+PATCH: /couriers/:courierId - update delivery profile info: toggle 'ready for work' status (active / not active), update 'current location' when active, update personal info
+
+GET: /orders/own?status=(new, completed, canceled) - get info about all orders details according to their status
```

Comment:
collision with endpoint for chef
if this one is for courier to check order he/she/it/they should deliver, may use `/deliveries/own` pattern

Reply:
Yeah, we probably didn't explain those endpoints correctly enough.
We assume an order will have two fields, orderStatus and deliveryStatus. For example, it could be like this:
orderStatus :
"Pending" - The order has been placed but has not been processed yet.
"Being Cooked" - The chef has accepted the order and started preparing the dish.
"Ready for Delivery" - The dish is prepared and ready for delivery.
"On Delivery" - The order is currently being delivered to the customer.
"Completed" - The order has been successfully delivered and completed.
"Canceled" - The order has been canceled by the customer or сhef.

deliveryStatus:
"Pending" - The order is awaiting assignment to a courier.
"On the way" - The courier is on the way to the delivery location.
"Delivered" - The order has been successfully delivered to the customer.
"Canceled" - The delivery was canceled by the courier

Thus, can we use such endpoints: 
GET /orders/user/:userId
GET /orders/user/:userId?status=(new,completed,canceled)
POST /order
GET, PATCH, DELETE /orders/user/:userId/:orderId

GET /orders/chef/:chefId
GET  /orders/chef/:chefId?status=(new,completed,canceled)
GET, PATCH  /orders/chef/:chefId/:orderId

GET /orders/courier/:courierId
GET  /orders/courier/:courierId?status=(new,completed,canceled)
GET, PATCH  /orders/courier/:courierId/:orderId

For an admin:
GET /orders 
GET, PATCH, DELETE /orders/:orderId

Reply:
then it's better to use
courier/:courierId/orders
chefs/:chefId/orders
users/:userId/orders
admins/:adminId/orders

as for me it's more self-explanatory and REST-like